auslib/db.py

```diff
@@ -1878,34 +1907,24 @@ def revokeRole(self, username, role, changed_by=None, old_data_version=None, tra
 
     def getPermission(self, username, permission, transaction=None):
         try:
-            row = self.select(where=[self.username == username, self.permission == permission], transaction=transaction)[0]
-            if row['options']:
-                row['options'] = json.loads(row['options'])
-            return row
+            return self.select(where=[self.username == username, self.permission == permission], transaction=transaction)[0]
         except IndexError:
             return {}
 
     def getUserPermissions(self, username, transaction=None):
         rows = self.select(columns=[self.permission, self.options, self.data_version], where=[self.username == username], transaction=transaction)
         ret = dict()
         for row in rows:
-            perm = row['permission']
-            opt = row['options']
-            ret[perm] = dict()
-            ret[perm]['data_version'] = row['data_version']
-            if opt:
-                ret[perm]['options'] = json.loads(opt)
-            else:
-                ret[perm]['options'] = None
+            ret[row["permission"]] = {
```

Comment:
I tried rewriting this as:
`return {row["permission"]: {"options": row["options"], "data_version": row["data_version"]} for row in rows}`

...but I actually think the current version is more readable.